fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Memo.java

```diff
@@ -151,8 +151,6 @@ private boolean needMerge(Group targetGroup, Group existedGroup) {
 
     // Merge srcGroup to dstGroup, srcGroup will be deleted
     private void mergeGroup(Group srcGroup, Group dstGroup) {
-        groups.remove(srcGroup);
```

Comment:
why not remove src group?

Reply:
I clean the group which will be merged, it will be clean by empty group

Reply:
removeOneGroup method is more complex and unnecessary. I don't think this change is better.

Reply:
could we remove the removeOneGroup method? if can not, this way is better